path/resolver/resolver.go

```diff
@@ -69,25 +69,42 @@ func (r *Resolver) ResolveToLastNode(ctx context.Context, fpath path.Path) (ipld
 	}
 
 	for len(p) > 0 {
-		val, rest, err := nd.Resolve(p)
+		lnk, rest, err := r.ResolveOnce(ctx, r.DAG, nd, p)
+		if lnk == nil {
```

Comment:
This'll drop the error, as far as I can tell.

Reply:
Ah, interesting. Can we make sure to document that explicitly? At a glance, it looks wrong, so let's make sure nobody comes along and 'fixes' it

Reply:
We should probably remove the error check below in that case.

Reply:
It can technically still be set at that point, but that would be a rather weird edge case